lnwallet/channel.go

```diff
@@ -4871,12 +4871,28 @@ func (lc *LightningChannel) ReceiveRevocation(revMsg *lnwire.RevokeAndAck) (
 		source, remoteChainTail, addUpdates, settleFailUpdates,
 	)
 
+	// We will soon be saving the current remote commitment to revocation
+	// log bucket, which is `lc.channelState.RemoteCommitment`. After that,
+	// the `RemoteCommitment` will be replaced with a newer version found
+	// in `CommitDiff`. Thus we need to compute the output indexes here
+	// before the change since the indexes are meant for the current,
+	// revoked remote commitment.
+	ourOutputIndex, theirOutputIndex, err := findOutputIndexesFromRemote(
```

Comment:
Why do we need this if their current commitment transaction is already stored within `remoteCommitChain`?

Reply:
These two indexes are not used or saved in the `ChannelCommitment`, rather, they are saved in the `RevocationLog` and is later used to create the to-local/remote outputs, where we create the spending outpoints.